src/SIGV4ConnectionConfig.test.tsx

```diff
@@ -0,0 +1,97 @@
+import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { SIGV4ConnectionConfig } from 'SIGV4ConnectionConfig';
+import { AwsAuthType } from 'types';
+
+const resetWindow = () => {
+  (window as any).grafanaBootData = {
+    settings: {
+      awsAllowedAuthProviders: [AwsAuthType.Credentials, AwsAuthType.Keys],
+      awsAssumeRoleEnabled: true,
+    },
+  };
+};
+
+describe('SIGV4ConnectionConfig', () => {
+  const setup = (onOptionsChange?: () => {}) => {
+    const props: DataSourcePluginOptionsEditorProps<any, any> = {
+      options: {
+        typeName: '',
+        id: 449,
+        uid: 'oIoBsD_Mz',
+        orgId: 1,
+        name: 'Elasticsearch',
+        type: 'elasticsearch',
+        typeLogoUrl: '',
+        access: 'proxy',
+        url: 'http://test.ts',
+        password: '',
+        user: '',
+        database: '',
+        basicAuth: false,
+        basicAuthUser: '',
+        basicAuthPassword: '',
+        withCredentials: false,
+        isDefault: false,
+        jsonData: {
+          esVersion: '7.10.0',
+          includeFrozen: false,
+          logLevelField: '',
+          logMessageField: '',
+          maxConcurrentShardRequests: 5,
+          sigV4AssumeRoleArn: 'arn:test',
+          sigV4Auth: true,
+          sigV4AuthType: 'credentials',
+          sigV4ExternalId: 'test-id',
+          sigV4Profile: 'profile',
+          sigV4Region: 'us-east-2',
+          timeField: '@timestamp',
+        },
+        secureJsonFields: { sigV4AccessKey: true, sigV4SecretKey: true },
```

Comment:
should these be filler strings instead of filler bools?

Reply:
For security reasons grafana backend doesn't pass the secure field as text to the frontend. It just passes a boolean value for each field to indicate weather it's set or not.